vnext/Microsoft.ReactNative.Cxx/JSI/JsiApiContext.h

```diff
@@ -20,6 +20,9 @@ facebook::jsi::Runtime &GetOrCreateContextRuntime(ReactContext const &context) n
 // The code is executed synchronously if it is already in JSDispatcher, or asynchronously otherwise.
 template <class TCodeWithRuntime>
 void ExecuteJsi(ReactContext const &context, TCodeWithRuntime const &code) {
+  if (context.Handle().SettingsSnapshot().UseWebDebugger()) {
```

Comment:
> if [](http://example.com/codeflow?start=1&length=3)

I do not think that this exception belongs here. There are many other ways how the JSI or JS Engine in general can be accessed. The issue should be addressed in the core code that works closely with the JS engine, rather than in the user code here.